plugin/trino-kafka-event-listener/src/main/java/io/trino/plugin/eventlistener/kafka/metadata/EnvMetadataProvider.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.plugin.eventlistener.kafka.metadata;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import static java.util.Objects.requireNonNull;
+
+public class EnvMetadataProvider
+        implements MetadataProvider
+{
+    protected final String prefix;
+
+    public EnvMetadataProvider(String prefix)
+    {
+        this.prefix = requireNonNull(prefix, "Prefix must be populated");
+    }
+
+    @Override
+    public Map<String, String> getMetadata()
+    {
+        return filter(System.getenv());
+    }
+
+    protected Map<String, String> filter(Map<String, String> metadata)
+    {
+        return metadata.entrySet().stream()
+                .filter(e -> e.getKey().startsWith(prefix))
+                .collect(Collectors.toMap(e -> e.getKey().replaceFirst(prefix, ""), Map.Entry::getValue));
```

Comment:
`replaceFirst` uses `regex` while `startsWith` uses plain text. Hence `replaceFirst` might not match what `startsWith` matched

Reply:
Good point, changed `replaceFirst` with `substring`